colossalai/shardformer/policies/opt.py

```diff
@@ -89,6 +95,27 @@ def module_policy(self):
                                                         policy=policy,
                                                         target_key=OPTDecoderLayer)
 
+        # use flash attention
+        if self.shard_config.enable_flash_attention:
+            policy[OPTAttention] = ModulePolicyDescription(
+                method_replacement={
+                    'forward': opt_flash_attention_forward,
+                },
+                sub_module_replacement=[
+                    SubModuleReplacementDescription(suffix="q_proj",
+                                                    target_module=Linear1D_Col,
+                                                    kwargs=dict(gather_output=True)),
+                    SubModuleReplacementDescription(suffix="k_proj",
+                                                    target_module=Linear1D_Col,
+                                                    kwargs=dict(gather_output=True)),
+                    SubModuleReplacementDescription(suffix="v_proj",
+                                                    target_module=Linear1D_Col,
+                                                    kwargs=dict(gather_output=True)),
+                    SubModuleReplacementDescription(suffix="out_proj",
+                                                    target_module=Linear1D_Col,
+                                                    kwargs=dict(gather_output=True)),
+                ])
```

Comment:
These have nothing to do with flash attention? They are in fact tensor parallelism.